### PR TITLE
Add E2E testing for Python and Update Java E2E with Latest Release

### PR DESCRIPTION
**Pull request by harrryr**
# Description of the issue
We need to update the release testing to point to branches containing the latest validator changes (`ga-release`, `ga-python`, `add-ec2-platform-support`) and also create E2E testing for Python artifacts.

The [aws-observability/aws-application-signals-test-framework](https://github.com/aws-observability/aws-application-signals-test-framework/) branches are used as follows:
- Java EKS: `ga-release`
- Java EC2 (nonASG): `ga-release`
- Java EC2 (ASG): `add-ec2-platform-support`
- Python EKS/EC2 (nonASG): `ga-python`
- Python EC2 (ASG): `add-ec2-platform-for-python-ga`

# Description of changes
Created new E2E workflow for Python release testing. ADOT artifacts were patched with release candidate ones to sync the artifacts. Also created new workflows to test EC2 with ASGs

Adot Artifacts used:
- Python EKS: [637423224110.dkr.ecr.us-east-1.amazonaws.com/aws-observability/adot-autoinstrumentation-python-staging:0.2.0-408d938](https://github.com/aws-observability/aws-otel-python-instrumentation/actions/runs/9120281390/job/25108995904)
- Python EC2: [s3://adot-main-build-staging-jar/aws_opentelemetry_distro-0.2.0-0408d938-py3-none-any.whl](https://github.com/aws-observability/aws-otel-python-instrumentation/actions/runs/9120281390/job/25108995603)
- Java EKS: [611364707713.dkr.ecr.us-west-2.amazonaws.com/adot-autoinstrumentation-java-operator-staging:1.33.0-SNAPSHOT-91cbba8](https://github.com/aws-observability/aws-otel-java-instrumentation/actions/runs/9119002183/job/25074698471)
- Java EC2: [s3://adot-main-build-staging-jar/aws-opentelemetry-agent.jar
](https://github.com/aws-observability/aws-otel-java-instrumentation/actions/runs/9119002183/job/25074697837)

Test run: https://github.com/aws/amazon-cloudwatch-agent/actions/runs/9132484921

Test run with missing metric aggregation added back into validation schemas: https://github.com/aws/amazon-cloudwatch-agent/actions/runs/9135922894

Rerun with all 6 e2e tests passing: https://github.com/aws/amazon-cloudwatch-agent/actions/runs/9136143337

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
_Describe what tests you have done._

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

File: .github/workflows/application-signals-python-e2e-ec2-test.yml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+env:
+  AWS_DEFAULT_REGION: us-east-1
+  APP_SIGNALS_E2E_TEST_ACCOUNT_ID: ${{ secrets.APP_SIGNALS_E2E_TEST_ACCOUNT_ID }}
+  SAMPLE_APP_ZIP: s3://aws-appsignals-sample-app-prod-us-east-1/python-sample-app.zip
+  METRIC_NAMESPACE: ApplicationSignals
+  LOG_GROUP_NAME: /aws/application-signals/data
+  TEST_RESOURCES_FOLDER: ${GITHUB_WORKSPACE}
+  GET_CW_AGENT_RPM_COMMAND: "aws s3 cp s3://${{ secrets.S3_INTEGRATION_BUCKET }}/integration-test/binary/${{ github.sha }}/amazon_linux/amd64/latest/amazon-cloudwatch-agent.rpm ./cw-agent.rpm"
+  GET_ADOT_WHEEL_COMMAND: "aws s3 cp s3://metric-schema-changes/aws_opentelemetry_distro-0.2.0-py3-none-any.whl ./aws_opentelemetry_distro-0.2.0-py3-none-any.whl && python3.9 -m pip install aws_opentelemetry_distro-0.2.0-py3-none-any.whl"
```
Comment (majanjua-amzn) [inline .github/workflows/application-signals-python-e2e-ec2-test.yml:23]: Double checking if this is correct

File: .github/workflows/application-signals-python-e2e-eks-test.yml [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+env:
+  AWS_DEFAULT_REGION: us-east-1
+  APP_SIGNALS_E2E_TEST_ACCOUNT_ID: ${{ secrets.APP_SIGNALS_E2E_TEST_ACCOUNT_ID }}
+  SAMPLE_APP_NAMESPACE: sample-app-namespace
+  SAMPLE_APP_FRONTEND_SERVICE_IMAGE: ${{ secrets.APP_SIGNALS_E2E_SAMPLE_APP_FRONTEND_SVC_IMG }}
+  SAMPLE_APP_REMOTE_SERVICE_IMAGE: ${{ secrets.APP_SIGNALS_E2E_SAMPLE_APP_REMOTE_SVC_IMG }}
+  METRIC_NAMESPACE: ApplicationSignals
+  LOG_GROUP_NAME: /aws/application-signals/data
+  ECR_INTEGRATION_TEST_REPO: "cwagent-integration-test"
+  APPLICATION_SIGNALS_ADOT_IMAGE: 637423224110.dkr.ecr.us-east-1.amazonaws.com/aws-observability/adot-autoinstrumentation-python-staging:0.2.0-408d938
```
Comment (majanjua-amzn) [inline .github/workflows/application-signals-python-e2e-eks-test.yml:28]: Double checking this as well.

File: .github/workflows/application-signals-java-e2e-eks-test.yml [hunk 3]
```diff
@@ -24,9 +24,10 @@ env:
   SAMPLE_APP_NAMESPACE: sample-app-namespace
   SAMPLE_APP_FRONTEND_SERVICE_IMAGE: ${{ secrets.APP_SIGNALS_E2E_SAMPLE_APP_FRONTEND_SVC_IMG }}
   SAMPLE_APP_REMOTE_SERVICE_IMAGE: ${{ secrets.APP_SIGNALS_E2E_SAMPLE_APP_REMOTE_SVC_IMG }}
-  METRIC_NAMESPACE: AppSignals
+  METRIC_NAMESPACE: ApplicationSignals
   LOG_GROUP_NAME: /aws/application-signals/data
   ECR_INTEGRATION_TEST_REPO: "cwagent-integration-test"
+  APPLICATION_SIGNALS_ADOT_IMAGE: 611364707713.dkr.ecr.us-west-2.amazonaws.com/adot-autoinstrumentation-java-operator-staging:1.33.0-SNAPSHOT-91cbba8
```
Comment (majanjua-amzn) [inline .github/workflows/application-signals-java-e2e-eks-test.yml:30]: Double checking this too

File: .github/workflows/application-signals-java-e2e-ec2-test.yml [hunk 4]
```diff
@@ -18,9 +18,9 @@ env:
   APP_SIGNALS_E2E_TEST_ACCOUNT_ID: ${{ secrets.APP_SIGNALS_E2E_TEST_ACCOUNT_ID }}
   SAMPLE_APP_FRONTEND_SERVICE_JAR: "s3://aws-appsignals-sample-app-prod-us-east-1/main-service.jar"
   SAMPLE_APP_REMOTE_SERVICE_JAR: "s3://aws-appsignals-sample-app-prod-us-east-1/remote-service.jar"
-  GET_ADOT_JAR_COMMAND: "wget -O adot.jar https://github.com/aws-observability/aws-otel-java-instrumentation/releases/latest/download/aws-opentelemetry-agent.jar"
+  GET_ADOT_JAR_COMMAND: "aws s3 cp s3://adot-main-build-staging-jar/aws-opentelemetry-agent.jar ./adot.jar"
```
[the remark this answers is not in the document]
Comment (harrryr) [reply .github/workflows/application-signals-java-e2e-ec2-test.yml:21]: Whenever ADOT Java is built by main-build workflow, the latest artifact is uploaded to the above bucket

File: .github/workflows/application-signals-java-e2e-ec2-test.yml [hunk 5]
```diff
@@ -31,7 +31,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           repository: aws-observability/aws-application-signals-test-framework
-          ref: main
+          ref: ga-release
```
[the remark this answers is not in the document]
Comment (harrryr) [reply .github/workflows/application-signals-java-e2e-ec2-test.yml:34]: No this is ec2 tests without asgs. `add-ec2-platform-support` is for ec2 test cases with asgs

File: .github/workflows/application-signals-python-e2e-ec2-test.yml (hunk 1 above)
Comment (harrryr) [reply .github/workflows/application-signals-python-e2e-ec2-test.yml:23]: GET_CW_AGENT_RPM_COMMAND: This is where the bucket the cw-agent artifact built by the run is stored
GET_ADOT_WHEEL_COMMAND: This is latest image built by @thp

Comment (majanjua-amzn) [reply .github/workflows/application-signals-python-e2e-ec2-test.yml:23]:
```suggestion
  GET_ADOT_WHEEL_COMMAND: "aws s3 cp s3://adot-main-build-staging-jar/aws_opentelemetry_distro-0.2.0-0408d938-py3-none-any.whl ./aws_opentelemetry_distro-0.2.0-0408d938-py3-none-any.whl && python3.9 -m pip install aws_opentelemetry_distro-0.2.0-0408d938-py3-none-any.whl"
```

Comment (majanjua-amzn) [reply .github/workflows/application-signals-python-e2e-ec2-test.yml:23]: Use the one that is created by the release workflow: https://github.com/aws-observability/aws-otel-python-instrumentation/actions/runs/9120281390/job/25108995603#step:11:23

Comment (majanjua-amzn) [reply .github/workflows/application-signals-python-e2e-ec2-test.yml:23]: @thpierce confirmed online that this WHL is an acceptable replacement. Resolving the comment

**Review by majanjua-amzn — APPROVED** [review]

**Pull request merged**